.cbmc-batch/jobs/aws_linked_list_swap_contents/aws_linked_list_swap_contents_harness.c

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <aws/common/linked_list.h>
+#include <proof_helpers/make_common_data_structures.h>
+
+void aws_linked_list_swap_contents_harness() {
+    /* data structure */
+    struct aws_linked_list a, b;
+
+    ensure_linked_list_is_allocated(&a, MAX_LINKED_LIST_ITEM_ALLOCATION);
+    ensure_linked_list_is_allocated(&b, MAX_LINKED_LIST_ITEM_ALLOCATION);
+
+    /* Keep the old first/last node of the linked lists */
+    struct aws_linked_list_node *old_a_head = &a.head;
```

Comment:
I was a bit confused why you need this when you're also saving ```a``` on the next line

Reply:
Because by saving `a` to `old_a` the address `&old_a.head` is now different from `&a.head`, whereas `old_a.tail.prev` still points to `&a.head`. So we need to save the address of the old head too. I will add a comment that clarifies that.